packages/kit-bg/src/vaults/impls/aptos/utils.ts

```
@@ -645,3 +494,36 @@ export function generateTransferCreateNft(
+
+export function decodeTxByBcsTxn(bcsTxn: string) {
+  const deserializer = new Deserializer(bufferUtils.hexToBytes(bcsTxn));
+  const simpleTxn = SimpleTransaction.deserialize(deserializer);
+  const rawTx = simpleTxn.rawTransaction;
+
+  let actionType = EDecodedTxActionType.UNKNOWN;
+  const payload = rawTx.payload;
+  switch (true) {
+    case payload instanceof TransactionPayloadEntryFunction:
+      // eslint-disable-next-line no-case-declarations
+      const functionName = payload.entryFunction.function_name.toString();
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)